addons/website/models/ir_ui_view.py

```diff
@@ -437,9 +437,10 @@ def _prepare_qcontext(self):
         """ Returns the qcontext : rendering context with website specific value (required
             to render website layout template)
         """
-        qcontext = super(View, self)._prepare_qcontext()
+        is_frontend_request = bool(request and getattr(request, 'is_frontend', False))
+        qcontext = super(View, self.with_context(no_qcontext_company=is_frontend_request))._prepare_qcontext()
```

Comment:
I thought about isolating `self.with_context(...)` on his own line but then we would have `super(View, self)` which is py-2 style super and I'm about sure one would (wrongly) remove `View, self` in the future. Keeping this way makes the line a bit long but will prove more robust to changes :+1: 

Is `no_qcontext_company` a new context-key ? In such case I would advocate we don't use a negative variable name but instead `with_qcontext_company` and a default `True` value (that would be set `False` here). We are trying to get rid of the "negative"-flags where we can.

Reply:
I actually removed that commit just before your review, as it was the only commit modifying something outside website, and introducing a new context key. 
I planed to add it only in master, to ease the current PR/Review, especially since it was not adding a lot of value for small pages, it only improved:
- non cached small pages
- cached & non cached bigger pages, as blog pages etc